src/app/newsletterSubscription/newsletterModal/newsletter.modal.js

```diff
@@ -0,0 +1,59 @@
+import angular from 'angular'
+import designationEditorService from 'common/services/api/designationEditor.service'
+import loading from 'common/components/loading/loading.component'
+
+import './newsletterModal.scss'
+
+const controllerName = 'newsletterModal'
+
+class NewsletterModalController {
+  /* @ngInject */
+  constructor (designationEditorService, designationNumber, displayName) {
+    this.designationEditorService = designationEditorService
+    this.designationNumber = designationNumber
+    this.displayName = displayName
+
+    this.step = 1
+    this.attributes = {
+      send_newsletter: 'Email'
+    }
+  }
+
+  next () {
+    if (this.step === 2) {
+      this.progress = true
+      this.designationEditorService.subscribeToNewsletter(this.designationNumber, this.attributes).then(() => {
+        // TODO: Handle success and failure
+        // Success
+        this.success = true
+      }, error => {
+        this.error = error
+      }).finally(() => {
+        this.step = 3
+        this.progress = false
+      })
+    } else {
+      this.step++
+    }
+  }
+
+  prev () {
+    this.step--
+    this.error = false
+  }
+
+  get includeEmailField () {
+    return ['Email', 'Both'].includes(this.attributes.send_newsletter)
```

Comment:
Should this function be removed since we always include it right now? Or are you leaving it here in case of Tatai changing things?

Reply:
I was leaving it in case we changed back